src/main/java/com/carrotsearch/junitbenchmarks/BenchmarkRule.java

```diff
@@ -113,4 +113,22 @@ private boolean workAroundWrappedStatements(Statement statement,
         field.set(object, new BenchmarkStatement(statement, description, consumers));
         return true;
     }
+
+    /**
+     * Uses reflection to get the {@code next} field, but falls back to {@code fNext} since
```

Comment:
The javadoc here is backwards from the code below.

We should change the code below: try to retrieve `next` in the `try` block, and then fall back to `fNext` only if `next` fails, as described.